azkaban-common/src/main/java/azkaban/imagemgmt/daos/ImageRampupDaoImpl.java

```diff
@@ -0,0 +1,555 @@
+   */
+  private void updateImageRampupPlanInternal(final ImageRampupPlanRequest imageRampupPlanRequest)
+      throws ImageMgmtException {
+    try {
+      final List<Object> params = new ArrayList<>();
+      final StringBuilder queryBuilder = new StringBuilder("update image_rampup_plan set ");
+      // As update is allowed only for active plan, if activatePlan is false then deactivate the
+      // current plan.
+      if (!imageRampupPlanRequest.isActivatePlan()) {
+        queryBuilder.append(" active = ?, ");
+        params.add(Boolean.valueOf(imageRampupPlanRequest.isActivatePlan()));
+      }
+      queryBuilder.append(" modified_by = ?, modified_on = ?");
+      params.add(imageRampupPlanRequest.getModifiedBy());
+      params.add(Timestamp.valueOf(LocalDateTime.now()));
+      queryBuilder.append(" where id = ? ");
+      params.add(imageRampupPlanRequest.getId());
+      this.databaseOperator.update(queryBuilder.toString(), Iterables.toArray(params, Object.class));
+    } catch (final SQLException ex) {
+      log.error("Exception while updating image version ", ex);
+      throw new ImageMgmtDaoException("Exception while updating image version");
+    }
+  }
+
+  /**
+   * This method up updates image rampup for an image type. The information such as version rampup
+   * percentage, stability tag etc. can be updated.
+   *
+   * @param imageRampupRequest
+   * @throws ImageMgmtException
+   */
+  private void updateImageRampup(final ImageRampupRequest imageRampupRequest)
+      throws ImageMgmtException {
+    try {
+      final List<Object> params = new ArrayList<>();
+      final StringBuilder queryBuilder = new StringBuilder("update image_rampup set ");
+      if (imageRampupRequest.getRampupPercentage() != null) {
+        queryBuilder.append(" rampup_percentage = ?, ");
+        params.add(imageRampupRequest.getRampupPercentage());
+      }
+      if (imageRampupRequest.getStabilityTag() != null) {
+        queryBuilder.append(" stability_tag = ?, ");
+        params.add(imageRampupRequest.getStabilityTag().getTagName());
+      }
+      queryBuilder.append(" modified_by = ?, modified_on = ?");
+      params.add(imageRampupRequest.getModifiedBy());
+      params.add(Timestamp.valueOf(LocalDateTime.now()));
+      queryBuilder.append(" where id = ? ");
+      params.add(imageRampupRequest.getId());
+      this.databaseOperator.update(queryBuilder.toString(), Iterables.toArray(params, Object.class));
+    } catch (final SQLException ex) {
+      log.error("Exception while updating image version ", ex);
+      throw new ImageMgmtDaoException("Exception while updating image version");
+    }
+  }
+
+  /**
+   * ResultSetHandler implementation class for fetching image rampup plan
+   */
+  public static class FetchImageRampupPlanHandler implements
+      ResultSetHandler<Optional<ImageRampupPlan>> {
+
+    @Override
+    public Optional<ImageRampupPlan> handle(final ResultSet rs) throws SQLException {
```

Comment:
Could you please point me to the code where the list of ImageRampup objects belonging to an image plan is added?